src/PubSub.ts

```diff
@@ -1,22 +1,53 @@
-import { PubSub as GPubSub } from '@google-cloud/pubsub';
-import { EmitOptions, ListenOptions } from './GoogleCloudPubSub';
+import { EmittedMessage, Metadata } from './EmittedMessage';
 
 /**
  * PubSub generic interface
  */
-export interface PubSub {
-  client: GPubSub;
-
+export interface PubSub<PSClient, PSSubscription, SubscriptionOptions> {
+  /**
+   * The exposed PubSub client
+   * If it is Google, then PSClient = PubSub
+   * https://github.com/googleapis/nodejs-pubsub/blob/master/src/pubsub.ts#L235
+   */
+  client: PSClient;
+  /**
+   * Exposed cached subscriptions
+   * If it is Google, then PSSubscription = Subscription
+   * https://github.com/googleapis/nodejs-pubsub/blob/master/src/subscription.ts#L211
+   */
+  subscriptions: Map<string, PSSubscription>;
   /**
    * Listen to a subscription
    * @param event Event to listen to
-   * @param handlers Message handler
+   * @param options Message handler
    */
-  listen<T>(event: string, options?: ListenOptions<T>): Promise<void>;
+  listen<T>(event: string, options?: ListenOptions<T, SubscriptionOptions>): Promise<void>;
```

Comment:
Since all your generics have meaningful names, you should rename `T` into something like `MessagePayload` too :)